apps/src/weblab/CdoBramble.js

```diff
@@ -455,6 +502,49 @@ export default class CdoBramble {
     }
   }
 
+  validateProjectChanged(callback) {
+    const startSourceFiles = this.api.getStartSources()?.files;
+    this.getAllFileData((err, userFiles) => {
+      // Don't block user if Bramble errors.
```

Comment:
Can you remind me what the scenario is for the user where we might hit this?

Reply:
there are 2 possible failures here:
1. bramble fails to read the list of entries (there's an `ls` command that lists all files on the filesystem)
2. bramble fails to read the file content from one of the entries

Reply:
And how will this show up to the user - looks like we won't block the user here, but will the still be able to use their project?

Reply:
yes, they'll still be able to use their project, although this function is only called when the user clicks the "finish" button (so the project will immediately unload anyways):
https://github.com/code-dot-org/code-dot-org/blob/504bcd5e62560b5a932c36a030ce018dc5a18f15/apps/src/weblab/WebLab.js#L348-L356

Reply:
(i'm merging this PR, but please do ask any other questions you have -- i'll have more clean-up/fixes to make before anything ships to users, so i can address any follow-ups there)